src/main/java/unicash/ui/SummaryWindow.java

```diff
@@ -69,28 +82,106 @@ public SummaryWindow() {
      *     </ul>
      */
     public void show(HashMap<String, Double> expenseSummary) {
-        Stage root = getRoot();
-        setPieChart(expenseSummary);
-        root.show();
+        if (!expenseSummary.isEmpty()) {
+            Stage root = getRoot();
+            root.show();
+            logger.fine("Showing UniCash summary page.");
+        }
     }
 
     /**
-     * Updates the pie chart with the data in model.expenseSummary
+     * Updates the pie chart with expense data
+     * Note that only MAX_DISPLAYED_CATEGORIES number of categories will be displayed on the pie chart
      */
     public void setPieChart(HashMap<String, Double> expenseSummary) {
+        pieChart.getData().clear();
 
-        ObservableList<PieChart.Data> pieChartData = FXCollections.observableArrayList();
-        for (HashMap.Entry<String, Double> entry : expenseSummary.entrySet()) {
-            String category = entry.getKey();
-            Double totalAmount = entry.getValue();
-            pieChartData.add(new PieChart.Data(category, totalAmount));
+        if (expenseSummary.isEmpty()) {
+            summaryMessage.setText(SUMMARY_MESSAGE_WHEN_NO_TRANSACTIONS);
+        } else {
+            summaryMessage.setText(SUMMARY_MESSAGE);
+            ObservableList<PieChart.Data> pieChartData = expenseSummary.entrySet().stream()
+                    .map(entry -> new PieChart.Data(entry.getKey(), entry.getValue()))
+                    .sorted(Comparator.comparing(PieChart.Data::getPieValue).reversed())
+                    .limit(MAX_DISPLAYED_CATEGORIES)
+                    .collect(Collectors.toCollection(FXCollections::observableArrayList));
+
+            pieChart.getData().setAll(pieChartData);
+            summaryMessageContainer.layout();
         }
+    }
 
-        pieChart.getData().clear();
+    /**
+     * Updates the line chart with expense data
+     * Note that only MAX_DISPLAYED_YEAR_MONTHS number of year-months will be displayed on the line chart
+     */
+    public void setLineGraph(HashMap<YearMonth, Double> expenseSummary) {
+        lineChart.getData().clear();
 
-        pieChart.getData().addAll(pieChartData);
+        expenseSummary = preprocessExpenseSummary(expenseSummary);
```

Comment:
Do we want to be modifying the argument directly? Or would it be better to create a new `HashMap<>()` for this preprocessed expense summary

Reply:
Yeah, missed this change you suggested in the previous review. Fixed it 😄